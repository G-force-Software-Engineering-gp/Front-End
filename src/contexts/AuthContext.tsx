import { createContext, useState, useEffect } from "react";
import jwt_decode from "jwt-decode";
import { useNavigate } from "react-router-dom";
import axios from "axios";
// import swal from "sweetalert";
// import axiosInstance from "../utils/axiosInstance";


export type AuthContextType = {
  user: any; // Replace with your user type
  setUser: React.Dispatch<React.SetStateAction<any>>;
  authTokens: any; // Replace with your auth tokens type
  setAuthTokens: React.Dispatch<React.SetStateAction<any>>;
  loginUser: (values: any) => Promise<void>;
  registerUser: (values: any) => Promise<void>;
  logoutUser: () => void;
};

const AuthContext = createContext<AuthContextType | null>(null);

export default AuthContext;
type AuthProviderProps = {
    children: React.ReactNode;
  };
  
  export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
  const [authTokens, setAuthTokens] = useState<any>(
    localStorage.getItem("authTokens")
      ? JSON.parse(localStorage.getItem("authTokens")!)
      : null
  );
  const [user, setUser] = useState<any>(
    localStorage.getItem("authTokens")
      ? jwt_decode(localStorage.getItem("authTokens")!)
      : null
  );
  const [loading, setloading] = useState(true);

  const Navigate = useNavigate();

  const loginUser = async (values: any) => {
    const response = await fetch(
        "https://amirmohammadkomijani.pythonanywhere.com/auth/jwt/create/",
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({
            email: values.email,
            password: values.password,
          }),
        }
      );
      const data = await response.json();
      console.log(data)
      if (response.status === 200) {
        setAuthTokens(data);
        setUser(jwt_decode(data.access));
        localStorage.setItem("authTokens", JSON.stringify(data));
        Navigate("/");
      } 
      else{
        console.log(data.error);
      }
  };
  //axios approach
        // await axios.post("https://amirmohammadkomijani.pythonanywhere.com/auth/jwt/create/",
        //     {
        //     email: values.email,
        //     password: values.password,
        //     },
        //     {
        //     headers: {
        //     'Content-Type': 'application/json',
        //     },
        //     })
        //     .then((response) => {
        //     const data = response.data;
        //     console.log(data);
        
        //     if (response.status === 200) {
        //         setAuthTokens(data);
        //         setUser(jwt_decode(data.access));
        //         localStorage.setItem('authTokens', JSON.stringify(data));
        //         Navigate('/');
        //     } else {
        //         console.log(data.error);
        //     }
        //     })
        //     .catch((error) => {
        //     console.error('API request failed:', error);
        //     });
        // }
  const registerUser = async (values: any) => {
      const reg_response = await fetch(
      "https://amirmohammadkomijani.pythonanywhere.com/auth/users/",
      {
          method: "POST",
          headers: {
          "Content-Type": "application/json",
          },
          body: JSON.stringify({
          first_name: values.firstName,
          last_name: values.lastName,
          email: values.email,
          username: values.username,
          password: values.password,
          }),
      });
      console.log(reg_response);
      const reg_data = await reg_response.json();
      if (reg_response.status === 201) {
        Navigate("/login");
      } else {
          console.log(reg_data.error);  
      }
  };

  //axios approach
        // await axios.post("https://amirmohammadkomijani.pythonanywhere.com/auth/users/",
        //     {
        //     first_name: values.firstName,
        //     last_name: values.lastName,
        //     email: values.email,
        //     username: values.username,
        //     password: values.password,
        //     },
        //     {
        //     headers: {
        //     'Content-Type': 'application/json',
        //     },
        //     })
        //     .then((reg_data) => {
        //         const data = reg_data.data;
        //         console.log(data);
        //         if (reg_data.status === 201) {
        //           Navigate('/login');
        //         } else {
        //           console.log(data.error);
        //         }
        //       })
        //       .catch((error) => {
        //         console.error('API request failed:', error);
        //       });
        // }

  const logoutUser = () => {
    setAuthTokens(null);
    setUser(null);
    localStorage.removeItem("authTokens");
    Navigate("/login");
  };

  const contextData: AuthContextType = {
    user,
    setUser,
    authTokens,
    setAuthTokens,
    loginUser,
    registerUser,
    logoutUser,
  };

  useEffect(() => {
    if (authTokens) {
        setUser(jwt_decode(authTokens.access));
      }
      setloading(false);
  }, [authTokens, loading]);

  return <AuthContext.Provider value={contextData}>{children}</AuthContext.Provider>;
};
