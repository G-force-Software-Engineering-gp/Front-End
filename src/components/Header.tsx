import {
  NavigationMenu,
  NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuTrigger,
  navigationMenuTriggerStyle,
} from '@/components/ui/navigation-menu';
import AuthContext from '@/contexts/AuthContext';
import { cn } from '@/lib/utils';
import AddWorkSapceHeader from '@/pages/HomePage/components/AddWorkSapceHeader';
import axios from 'axios';
import { BellDot, HelpCircle, User2 } from 'lucide-react';
import React, { useContext, useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { Input } from './ui/input';
import { ModeToggle } from './ui/mode-toggle';

const ListItem = React.forwardRef<React.ElementRef<'a'>, React.ComponentPropsWithoutRef<'a'>>(
  ({ className, title, children, ...props }, ref) => {
    return (
      <li>
        <NavigationMenuLink asChild>
          <a
            ref={ref}
            className={cn(
              'block select-none space-y-1 rounded-md p-3 leading-none no-underline outline-none transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground',
              className
            )}
            {...props}
          >
            <div className="text-sm font-medium leading-none">{title}</div>
            <p className="line-clamp-2 text-sm leading-snug text-muted-foreground">{children}</p>
          </a>
        </NavigationMenuLink>
      </li>
    );
  }
);
ListItem.displayName = 'ListItem';

const Header = () => {
  const navigate = useNavigate();
  const components: { title: string; href: string; description: string }[] = [
    {
      title: 'Alert Dialog',
      href: '/docs/primitives/alert-dialog',
      description: 'A modal dialog that interrupts the user with important content and expects a response.',
    },
    {
      title: 'Hover Card',
      href: '/docs/primitives/hover-card',
      description: 'For sighted users to preview content available behind a link.',
    },
    {
      title: 'Progress',
      href: '/docs/primitives/progress',
      description:
        'Displays an indicator showing the completion progress of a task, typically displayed as a progress bar.',
    },
    {
      title: 'Scroll-area',
      href: '/docs/primitives/scroll-area',
      description: 'Visually or semantically separates content.',
    },
    {
      title: 'Tabs',
      href: '/docs/primitives/tabs',
      description: 'A set of layered sections of content—known as tab panels—that are displayed one at a time.',
    },
    {
      title: 'Tooltip',
      href: '/docs/primitives/tooltip',
      description:
        'A popup that displays information related to an element when the element receives keyboard focus or the mouse hovers over it.',
    },
  ];
  let authTokens = useContext(AuthContext)?.authTokens;
  const [workspaces, setworkspaces] = useState<any[]>([]);
  const gettingData = async () => {
    const { data } = await axios
      .get('https://amirmohammadkomijani.pythonanywhere.com/tascrum/workspace/', {
        headers: {
          Authorization: `JWT ${authTokens?.access}`,
        },
      })
      .then((response) => response);
    setworkspaces(data);
  };
  useEffect(() => {
    gettingData();
  }, []);
  return (
    <div data-testid="header">
      <>
        <div className="hidden flex-col border-b-2 lg:flex">
          <div className="flex-1 p-2 px-5 ">
            <div className="flex items-center justify-between">
              <div className="text-2xl font-bold">
                <>
                  <NavigationMenu>
                    <NavigationMenuList>
                      <NavigationMenuItem>
                        <span className="cursor-pointer">
                          <NavigationMenuLink
                            className={navigationMenuTriggerStyle()}
                            onClick={() => {
                              navigate('/');
                            }}
                          >
                            Logo and Name
                          </NavigationMenuLink>
                        </span>
                      </NavigationMenuItem>

                      <NavigationMenuItem>
                        <NavigationMenuTrigger>Workspaces</NavigationMenuTrigger>
                        <NavigationMenuContent>
                          <ul className="grid gap-3 p-6 md:w-[400px] lg:w-[500px] lg:grid-cols-[.75fr_1fr]">
                            <li className="row-span-3">
                              <NavigationMenuLink asChild>
                                <a
                                  className="flex h-full w-full select-none flex-col justify-end rounded-md bg-gradient-to-b from-muted/50 to-muted p-6 no-underline outline-none focus:shadow-md"
                                  href="/"
                                >
                                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" className="h-6 w-6">
                                    <rect width="256" height="256" fill="none"></rect>
                                    <line
                                      x1="208"
                                      y1="128"
                                      x2="128"
                                      y2="208"
                                      fill="none"
                                      stroke="currentColor"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      strokeWidth="16"
                                    ></line>
                                    <line
                                      x1="192"
                                      y1="40"
                                      x2="40"
                                      y2="192"
                                      fill="none"
                                      stroke="currentColor"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      strokeWidth="16"
                                    ></line>
                                  </svg>
                                  <div className="mb-2 mt-4 text-lg font-medium">Workspaces</div>
                                  <p className="text-sm leading-tight text-muted-foreground">
                                    You can see list of all your workspaces here.
                                  </p>
                                </a>
                              </NavigationMenuLink>
                            </li>
                            <li>
                              {workspaces
                                ? workspaces.map((item) => <ListItem title={item.name}>{item.description}</ListItem>)
                                : ''}
                            </li>
                          </ul>
                        </NavigationMenuContent>
                      </NavigationMenuItem>

                      <NavigationMenuItem>
                        <NavigationMenuTrigger>Recent</NavigationMenuTrigger>
                        <NavigationMenuContent>
                          <ul className="grid gap-3 p-6 md:w-[400px] lg:w-[500px] lg:grid-cols-[.75fr_1fr]">
                            <li className="row-span-3">
                              <NavigationMenuLink asChild>
                                <a
                                  className="flex h-full w-full select-none flex-col justify-end rounded-md bg-gradient-to-b from-muted/50 to-muted p-6 no-underline outline-none focus:shadow-md"
                                  href="/"
                                >
                                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256" className="h-6 w-6">
                                    <rect width="256" height="256" fill="none"></rect>
                                    <line
                                      x1="208"
                                      y1="128"
                                      x2="128"
                                      y2="208"
                                      fill="none"
                                      stroke="currentColor"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      strokeWidth="16"
                                    ></line>
                                    <line
                                      x1="192"
                                      y1="40"
                                      x2="40"
                                      y2="192"
                                      fill="none"
                                      stroke="currentColor"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      strokeWidth="16"
                                    ></line>
                                  </svg>
                                  <div className="mb-2 mt-4 text-lg font-medium">Recent Workspaces</div>
                                  <p className="text-sm leading-tight text-muted-foreground">
                                    You can see a list of your recent workspaces
                                  </p>
                                </a>
                              </NavigationMenuLink>
                            </li>
                            <ListItem title="Recent 1">You can see your recent workspace</ListItem>
                            <ListItem title="Recent 1">You can see your recent workspace</ListItem>
                            <ListItem title="Recent 1">You can see your recent workspace</ListItem>
                          </ul>
                        </NavigationMenuContent>
                      </NavigationMenuItem>

                      <NavigationMenuItem>
                        <NavigationMenuTrigger>Templates</NavigationMenuTrigger>
                        <NavigationMenuContent>
                          <ul className="grid w-[400px] cursor-pointer gap-3 p-4 md:w-[500px] md:grid-cols-2 lg:w-[600px] ">
                            {components.map((component) => (
                              <ListItem key={component.title} title={component.title}>
                                {component.description}
                              </ListItem>
                            ))}
                          </ul>
                        </NavigationMenuContent>
                      </NavigationMenuItem>

                      <AddWorkSapceHeader />
                    </NavigationMenuList>
                  </NavigationMenu>
                </>
              </div>
              <div className="flex items-center space-x-2">
                <Input type="text" placeholder="Search" />
                <ModeToggle data-testid="mode-toggle11" />
                <BellDot className="h-9 w-9" />
                <HelpCircle className="h-9 w-9" />
                <User2 data-testid="user-icon" className="h-9 w-9" onClick={() => navigate('/settings')} />
              </div>
            </div>
          </div>
        </div>
      </>
    </div>
  );
};

export default Header;
