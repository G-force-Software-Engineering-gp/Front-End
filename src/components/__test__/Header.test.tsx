import { AuthProvider } from '@/contexts/AuthContext';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header'; // Import your Header component

describe('Header Component', () => {
  test('should render without errors', () => {
    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
  });

  test('Header component matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Clicking the logo navigates to the home page', () => {
    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    const LogoElement = screen.getByText('Logo and Name');
    fireEvent.click(LogoElement);
  });

  test('Clicking the User2 icon navigates to the settings page', () => {
    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    const LogoElement = screen.getByTestId('user-icon');
    fireEvent.click(LogoElement);
  });

  test('Typing into the search input updates its value', () => {
    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    const input = screen.getByPlaceholderText('Search') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'Test Search' } });
    expect(input.value).toBe('Test Search');
  });

  test('Hides user information when not authenticated', () => {
    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    expect(screen.queryByText('User2')).toBeNull();
  });
  test('Displays workspaces correctly', async () => {
    const mockWorkspaces = [
      { name: 'Workspace1', description: 'Description1' },
      { name: 'Workspace2', description: 'Description2' },
    ];

    jest.mock('axios', () => ({
      get: () => ({ data: mockWorkspaces }),
    }));

    render(
      <MemoryRouter>
        <AuthProvider>
          <Header />
        </AuthProvider>
      </MemoryRouter>
    );
    waitFor(() => {
      mockWorkspaces.forEach((workspace) => {
        expect(screen.getByText(workspace.name)).toBeInTheDocument();
        expect(screen.getByText(workspace.description)).toBeInTheDocument();
      });
    });
  });
});
