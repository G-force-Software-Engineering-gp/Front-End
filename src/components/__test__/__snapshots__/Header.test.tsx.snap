// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="header"
  >
    <div
      class="hidden flex-col border-b-2 lg:flex"
    >
      <div
        class="flex-1 p-2 px-5 "
      >
        <div
          class="flex items-center justify-between"
        >
          <div
            class="text-2xl font-bold"
          >
            <nav
              aria-label="Main"
              class="relative z-10 flex max-w-max flex-1 items-center justify-center"
              data-orientation="horizontal"
              dir="ltr"
            >
              <div
                style="position: relative;"
              >
                <ul
                  class="group flex flex-1 list-none items-center justify-center space-x-1"
                  data-orientation="horizontal"
                  dir="ltr"
                >
                  <li>
                    <span
                      class="cursor-pointer"
                    >
                      <a
                        class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50"
                        data-radix-collection-item=""
                      >
                        Logo and Name
                      </a>
                    </span>
                  </li>
                  <li>
                    <button
                      aria-controls="radix-:ra:-content-radix-:rc:"
                      aria-expanded="false"
                      class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50 group"
                      data-radix-collection-item=""
                      data-state="closed"
                      id="radix-:ra:-trigger-radix-:rc:"
                    >
                      Workspaces 
                      <svg
                        aria-hidden="true"
                        class="relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="6 9 12 15 18 9"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-controls="radix-:ra:-content-radix-:rd:"
                      aria-expanded="false"
                      class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50 group"
                      data-radix-collection-item=""
                      data-state="closed"
                      id="radix-:ra:-trigger-radix-:rd:"
                    >
                      Recent 
                      <svg
                        aria-hidden="true"
                        class="relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="6 9 12 15 18 9"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-controls="radix-:ra:-content-radix-:re:"
                      aria-expanded="false"
                      class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50 group"
                      data-radix-collection-item=""
                      data-state="closed"
                      id="radix-:ra:-trigger-radix-:re:"
                    >
                      Templates 
                      <svg
                        aria-hidden="true"
                        class="relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="6 9 12 15 18 9"
                        />
                      </svg>
                    </button>
                  </li>
                  <button
                    aria-controls="radix-:rf:"
                    aria-expanded="false"
                    aria-haspopup="dialog"
                    class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 !bg-blue-500"
                    data-state="closed"
                    data-testid="clicking"
                    type="button"
                  >
                    Create
                  </button>
                </ul>
              </div>
              <div
                class="absolute left-0 top-full flex justify-center"
              />
            </nav>
          </div>
          <div
            class="flex items-center space-x-2"
          >
            <input
              class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
              placeholder="Search"
              type="text"
            />
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
              data-state="closed"
              id="radix-:ri:"
              type="button"
            >
              <svg
                class="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <path
                  d="M12 2v2"
                />
                <path
                  d="M12 20v2"
                />
                <path
                  d="m4.93 4.93 1.41 1.41"
                />
                <path
                  d="m17.66 17.66 1.41 1.41"
                />
                <path
                  d="M2 12h2"
                />
                <path
                  d="M20 12h2"
                />
                <path
                  d="m6.34 17.66-1.41 1.41"
                />
                <path
                  d="m19.07 4.93-1.41 1.41"
                />
              </svg>
              <svg
                class="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
                />
              </svg>
              <span
                class="sr-only"
              >
                Toggle theme
              </span>
            </button>
            <svg
              class="h-9 w-9"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.4 14.9C20.2 16.4 21 17 21 17H3s3-2 3-9c0-3.3 2.7-6 6-6 .7 0 1.3.1 1.9.3"
              />
              <path
                d="M10.3 21a1.94 1.94 0 0 0 3.4 0"
              />
              <circle
                cx="18"
                cy="8"
                r="3"
              />
            </svg>
            <svg
              class="h-9 w-9"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
              />
              <path
                d="M12 17h.01"
              />
            </svg>
            <svg
              class="h-9 w-9"
              data-testid="user-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="8"
                r="5"
              />
              <path
                d="M20 21a8 8 0 1 0-16 0"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
